builtin/common/chatcommands.lua

```diff
@@ -27,4 +27,76 @@ function core.override_chatcommand(name, redefinition)
 		rawset(chatcommand, k, v)
 	end
 	core.registered_chatcommands[name] = chatcommand
-end
\ No newline at end of file
+end
+
+local cmd_marker = "/"
+
+if INIT == "client" then
+	cmd_marker = "."
+end
+
+local function priv_cmd(name, param)
+	local function format_help_line(cmd, def)
+		local msg = core.colorize("#00ffff", cmd_marker .. cmd)
+		if def.params and def.params ~= "" then
+			msg = msg .. " " .. def.params
+		end
+		if def.description and def.description ~= "" then
+			msg = msg .. ": " .. def.description
+		end
+		return msg
+	end
+	if param == "" then
+		local msg = ""
+		local cmds = {}
+		for cmd, def in pairs(core.registered_chatcommands) do
+			if INIT == "client" or core.check_player_privs(name, def.privs) then
+				cmds[#cmds + 1] = cmd
+			end
+		end
+		table.sort(cmds)
+		return true, "Available commands: " .. table.concat(cmds, " ") .. "\n"
+				.. "Use '"..cmd_marker.."help <cmd>' to get more information,"
+				.. " or '"..cmd_marker.."help all' to list everything."
+	elseif param == "all" then
+		local cmds = {}
+		for cmd, def in pairs(core.registered_chatcommands) do
+			if INIT == "client" or core.check_player_privs(name, def.privs) then
+				cmds[#cmds + 1] = format_help_line(cmd, def)
+			end
+		end
+		table.sort(cmds)
+		return true, "Available commands:\n"..table.concat(cmds, "\n")
+	elseif INIT == "game" and param == "privs" then
```

Comment:
"all" stores the result of `format_help_line` in `cmds` not the name of the command.